### ISSUE TITLE
Unable to mount existing volume to server: invalid input in field 'volumes'

### ISSUE THREAD

**Issue by TLINDEN** [ansible-collections/hetzner.hcloud#585]
##### SUMMARY
I am trying to attach an existing volume to a server during server creation using the `volumes:` key, which fails with some incomprehensible error.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- hetzner.hcloud.server

##### ANSIBLE VERSION

```paste below
ansible --version
ansible [core 2.16.3]
  config file = /home/scip/dev/bsdnix/ansible.cfg
  configured module search path = ['/home/scip/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/scip/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
ansible-galaxy collection list hetzner.hcloud

# /usr/lib/python3/dist-packages/ansible_collections
Collection     Version
-------------- -------
hetzner.hcloud 2.4.1  
```

##### CONFIGURATION

```paste below
ansible-config dump --only-changed
CONFIG_FILE() = /home/scip/dev/bsdnix/ansible.cfg
DEFAULT_STDOUT_CALLBACK(/home/scip/dev/bsdnix/ansible.cfg) = yaml
DEFAULT_VAULT_PASSWORD_FILE(/home/scip/dev/bsdnix/ansible.cfg) = /home/scip/.config/ansible/hcloud.secret
EDITOR(env: EDITOR) = vim
HOST_KEY_CHECKING(/home/scip/dev/bsdnix/ansible.cfg) = False
INVENTORY_ENABLED(/home/scip/dev/bsdnix/ansible.cfg) = ['hcloud', 'host_list', 'yaml']
```

##### OS / ENVIRONMENT
I'm working on kubuntu 24.04.

##### STEPS TO REPRODUCE
I have one role, which creates the volume, which works.

The next role creates the server, which works as well unless I add the `volumes:` field.

```yaml
---
- name: Create server
  hetzner.hcloud.server:
    name: "{{ hostname }}"
    server_type: "{{ type }}"
    image: "{{ image }}"
    location: "{{ location }}"
    enable_ipv4: false
    state: present
    ssh_keys: "{{ ssh_keys }}"
    volumes:
      - "{{ storage.home.name }}"
  register: server
```

The variable `storage.home.name` resolves properly, the same variable works like a charm during volume creation.

##### EXPECTED RESULTS
Just create the server with the volume attached.

##### ACTUAL RESULTS

```paste below
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_hetzner.hcloud.server_payload_j4wr_4ot/ansible_hetzner.hcloud.server_payload.zip/ansible_collect
ions/hetzner/hcloud/plugins/modules/server.py", line 471, in _create_server
    resp = self.client.servers.create(**params)                                                                      
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                      
  File "/tmp/ansible_hetzner.hcloud.server_payload_j4wr_4ot/ansible_hetzner.hcloud.server_payload.zip/ansible_collect
ions/hetzner/hcloud/plugins/module_utils/vendor/hcloud/servers/client.py", line 641, in create
    response = self._client.request(url="/servers", method="POST", json=data)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/ansible_hetzner.hcloud.server_payload_j4wr_4ot/ansible_hetzner.hcloud.server_payload.zip/ansible_collect
ions/hetzner/hcloud/plugins/module_utils/vendor/hcloud/_client.py", line 231, in request
    self._raise_exception_from_content(content)
  File "/tmp/ansible_hetzner.hcloud.server_payload_j4wr_4ot/ansible_hetzner.hcloud.server_payload.zip/ansible_collect
ions/hetzner/hcloud/plugins/module_utils/vendor/hcloud/_client.py", line 185, in _raise_exception_from_content 
    raise APIException(                                                                                              
ansible_collections.hetzner.hcloud.plugins.module_utils.vendor.hcloud._exceptions.APIException: invalid input in fiel
d 'volumes'                                               
fatal: [shell]: FAILED! => changed=false                                                                             
  ansible_facts:       
    discovered_interpreter_python: /usr/bin/python3
  failure:
    code: invalid_input                                   
    details:                                                                                                         
      fields:                                                                                                        
      - messages:                                                                                                    
        - Not a valid integer.
        name: volumes.0                                   
    message: invalid input in field 'volumes'                                                                        
  invocation:
    module_args:                                          
      allow_deprecated_image: false
      api_endpoint: https://api.hetzner.cloud/v1
      api_token: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
      backups: null
      datacenter: null                                    
      delete_protection: null                    
      enable_ipv4: false                                  
      enable_ipv6: true                                                                                              
      firewalls: null                                                                                                
      force: false       
      force_upgrade: null                                                                                            
      id: null                                                                                                       
      image: '198468958'
      ipv4: null                                          
      ipv6: null
      firewalls: null                                                                                                
      force: false       
      force_upgrade: null                                                                                            
      id: null                                                                                                       
      image: '198468958'
      ipv4: null                                          
      ipv6: null
      labels: null
      location: fsn1
      name: shell.daemon.de
      placement_group: null
      private_networks: null
      rebuild_protection: null
      rescue_mode: null
      server_type: cx22
      ssh_keys:
      - scip@e3
      - scip@tripod
      - scip@pixel8
      state: present
      upgrade_disk: false
      user_data: null
      volumes:
      - zhcloud
  msg: invalid input in field 'volumes'
```

What's especially irritating is that it says "Not a valid integer.", really weird. Looks like some slice iteration error.

**Comment by TLINDEN** [comment]
The API expects a volume id at this point:

> volumes
> array of integers (int64)
> Volume IDs which should be attached to the Server at the creation time. Volumes must be in the same Location.

However, the ansible documentation of the hetzner.hcloud.server module says it wants a name. So, I suspect I need to resolve the name to the ID somehow. But how?

**Comment by TLINDEN** [comment]
But I can just attach the volume using the hetzner.hcloud.volume module:

```yaml
- name: Attach the home volume
  hetzner.hcloud.volume:
    name: "{{ storage.home.name }}"
    server: "{{ hostname }}"
    state: present
```

So, here it accepts the name. Strange.

**Comment by jooola** [comment]
Please upgrade your hetzner.hcloud collection, this has been solved in 7fdefcfa0243b84a3edb58566ec710e4f4a6db8d
which was released in v3.1.0 https://github.com/ansible-collections/hetzner.hcloud/blob/main/CHANGELOG.rst#v3-1-0
